### PR TITLE
Fix - The mobile menu does not constrain focus when a fallback is used

### PR DESCRIPTION
**Pull request by acosmin**
Fix for #832

### CODE REVIEW

**Comment by carolinan** [comment]
The focus does not stay within the search modal when it is open.
This means that when you tab backwards you end up below the modal and you can't see where you are.

**Comment by acosmin** [comment]
@carolinan give it a try now :)

**Comment by acosmin** [comment]
I think the script can be simplified, I'll take a look

**Comment by acosmin** [comment]
@carolinan Refactored 💯 😃

**Comment by acosmin** [comment]
Finally 😄  adding so many menu locations is a pain from a code point of view, I'm starting to regret voting for this solution 😆

File: assets/js/index.js [hunk 1]
```diff
@@ -435,74 +435,29 @@ twentytwenty.modalMenu = {
...
-			lastModalMenuItems = hasSocialMenu ? socialLinks : menuLinks;
-			focusedElementParentLi = twentytwentyFindParents( event.target, 'li' );
-			focusedElementIsInsideModal = twentytwentyFindParents( event.target, '.menu-modal' ).length > 0;
+				elements = modal.querySelectorAll( 'input, a, button' );
+				elements = Array.prototype.slice.call( elements );
 
-			lastMenuItem = lastModalMenuItems[lastModalMenuItems.length - 1];
+				lastEl = elements[ elements.length - 1 ];
+				firstEl = elements[0];
+				activeEl = document.activeElement;
```
Comment (c4rl0sbr4v0) [inline assets/js/index.js]: Are these variables taking into account that the last menu item can have submenus open or closed?

Comment (acosmin) [reply assets/js/index.js]: I hope so 😆

Comment (acosmin) [reply assets/js/index.js]: You're right about that, I'll have to fix it 😄

Comment (acosmin) [reply assets/js/index.js]: Fixed in 6a9fec2

Please check it if you have the time 😄

Comment (c4rl0sbr4v0) [reply assets/js/index.js]: You are making a huuuge refactor 😆 , let you know if fails.

Comment (c4rl0sbr4v0) [reply assets/js/index.js]: Works like a charm :-)

Comment (acosmin) [reply assets/js/index.js]: thanks 😄 marking this as resolved

**Comment by carolinan** [comment]
I have tested in Chrome, Firefox, IE11 and Edge, also on 4.9 and on android mobile and Ipad (Chrome, Safari). I have limited experience with using screen readers but I tested with NVDA and it seems to work well.

**Pull request merged**